packages/core-ethereum/src/indexer/index.ts

```diff
@@ -86,9 +87,30 @@ class Indexer extends EventEmitter {
     this.chain.subscribeBlock((b) => {
       this.onNewBlock(b)
     })
+
     this.chain.subscribeError((error: any) => {
       log(chalk.red(`etherjs error: ${error}`))
-      this.restart()
+      // if provider connection issue
+      if (
+        [error?.code, String(error)].includes(errors.SERVER_ERROR) ||
+        [error?.code, String(error)].includes(errors.TIMEOUT) ||
+        [error?.code, String(error)].includes('ECONNRESET')
```

Comment:
Not really elegant, can use rephrase the check to reduce duplication?

Reply:
Updated check with `.some()`